sherpa/fit.py

```diff
@@ -1000,23 +998,22 @@ class Fit(NoNewAttributesAfterInit):
 
     Parameters
     ----------
-    data : sherpa.data.Data or sherpa.data.DataSimulFit instance
+    data : `sherpa.data.Data` or `sherpa.data.DataSimulFit`
        The data to be fit.
-    model : sherpa.models.model.Model or sherpa.models.model.SimulFitModel instance
-       The model to fit to the data. It should match the `data` parameter
-       (i.e. be a SimulFitModel object when data is a DataSimulFit).
-    stat : sherpa.stats.Stat instance or None, optional
+    model : `sherpa.models.model.Model` or `sherpa.models.model.SimulFitModel`
+       The model to fit to the data. It should match the ``data`` parameter
+       (i.e. be a `SimulFitModel` object when data is a `DataSimulFit`).
+    stat : `sherpa.stats.Stat` or `None`, optional
        The statistic object to use. If not given then
-       :py:class:`~sherpa.stats.Chi2Gehrels` is used.
+       `Chi2Gehrels` is used.
     method : sherpa.optmethods.OptMethod instance or None, optional
-       The optimiser to use. If not given then
-       :py:class:`~sherpa.optmethods.LevMar` is used.
+       The optimiser to use. If not given then `LevMar` is used.
```

Comment:
Although `LevMar` is now rendered as a "clickable link" at https://sherpa--1268.org.readthedocs.build/en/1268/fit/api/sherpa.fit.Fit.html it doesn't actually work - ie it doesn't link to anything. Actually, I note that this holds for other terms that are used in this section that are not defined in this module (e.g. `Chi2Gehrels`, `Covariance`, `SimulFitModel`). I think you'd need to give the full path - e.g. "`sherpa.optmethods.LevMar`" .

Reply:
Yeah, I was experimenting. I can, however, do "~sherpa.optmethods.LevMar" , which will render as "LevMar" in the html, but still show the full link in the docstring itself obviously.
Which version do we prefer?

(Eventually we can turn sphinx warnings into errors which will catch cases like that, but for now we have too many other warnings.)

Reply:
So, "You need to `~sherpa.frobnicate.bar` to get the answer" is a lot less messy than the old "You need to :py:class:`~sherpa.frobnicate.bar` to get the answer" so I think in general we should go for the linkable version (so `~sherpa.optmethods.LevMar`), but I don't think it's a hard-and-fast rule. Does this seem workable?

Out of interest, is it

    ~sherpa.optmethods.LevMar
    `~sherpa.optmethods.LevMar`

ie do you need the back-ticks?

Reply:
Yes, you do. (Exept for some very special cases, e.g. the None in a parameter, or a list of numpy specifics (like "array-like"). I think some of those speical rules also work better with numpydoc, which is like slightly better than napoleon for it's formatting, but that's a different issue.